### PR TITLE
This commit fixes port 3030: Connection refused with seth-rpc

### PR DESCRIPTION
**Pull request by cliveb**
Date; Fri July 6, 17:22:23 2018 -0800

	This commit fixes port 3030: Connection refused with seth-rpc
	reported as STL-1300
	[](https://jira.hyperledger.org/browse/STL-1300)

Signed-off-by: clive boulton <clive.boulton@gmail.com>

### CODE REVIEW

**Comment by peterschwarz** [comment]
Thanks for the submission.  Could you please amend your commit message to follow more of the guidelines found in this [blog post](https://chris.beams.io/posts/git-commit/)?   Sawtooth projects, as a whole, use this format (see the [core contributor's guide](https://sawtooth.hyperledger.org/docs/core/nightly/master/community/contributing.html#the-commit-process))

File: docker-compose-installed.yaml [hunk 1]
```diff
@@ -21,7 +21,11 @@ services:
       context: .
       dockerfile: ./cli/Dockerfile-installed
       args:
-        ISOLATION_ID: ${ISOLATION_ID}
+        args:
```
Comment (dplumb94) [inline docker-compose-installed.yaml]: Adding these proxy args should be a different commit and PR, since this change is not related to the bug fix.

**Comment by rberg2** [comment]
It looks like the tests are not passing on this one.

```
[jenkins-Sawtooth-Hyperledger-sawtooth-seth-PR-43-3] Running shell script
+ docker-compose -f docker-compose-installed.yaml build
The Compose file './docker-compose-installed.yaml' is invalid because:
services.seth-cli.build.args contains ["ISOLATION_ID=5c9c761e2ed01cfbe220f29a8ad2a834232776c50a82f00d44c8307e96b31558", "http_proxy", "https_proxy", "no_proxy"], which is an invalid type, it should be a string, number, or a null
services.seth-tp.build.args contains ["ISOLATION_ID=5c9c761e2ed01cfbe220f29a8ad2a834232776c50a82f00d44c8307e96b31558", "http_proxy", "https_proxy", "no_proxy"], which is an invalid type, it should be a string, number, or a null

```

File: docker-compose-installed.yaml [hunk 2]
```diff
@@ -115,15 +123,20 @@ services:
...
+        - http_proxy
+        - https_proxy
+        - no_proxy
     image: sawtooth-seth-rpc:${ISOLATION_ID}
     container_name: seth-rpc
     volumes:
       - ./build/debs:/build
     depends_on:
       - validator
+    expose:
```
Comment (dplumb94) [inline docker-compose-installed.yaml]: The `ports` option publishes port 3030 to the host machine, which we need in order to send requests from the host machine. `expose` will make port 3030 available to other containers in the docker network, but this is already implied by the `ports` option.

File: docker-compose-installed.yaml [hunk 3]
```diff
@@ -115,15 +123,20 @@ services:
...
+      - 3030    
     ports:
-      - 3030:3030
+      - '3030:3030'
```
Comment (dplumb94) [inline docker-compose-installed.yaml]: Was this causing any issues? Though it is a best practice to configure ports as a string in docker-compose, this should be a different commit and PR if it does not relate to the bug being fixed here.

File: docker-compose-installed.yaml (hunk 1 above)
Comment (cliveb) [reply docker-compose-installed.yaml]: Ok, I will double check. 

I'm pretty sure our team was unable to standup the current sawtooth-seth repo without ISOLATION_ID: ${ISOLATION_ID} added to docker-compose-installed.yaml after docker-compose.yaml fixed for port 3030 connection refused in Seth issue.

Possibly due to docker 18.03.1-ce changes landing. https://docs.docker.com/release-notes/docker-ce/#stable-releases

File: docker-compose-installed.yaml (hunk 2 above)
Comment (cliveb) [reply docker-compose-installed.yaml]: Thanks. I will remove expose and test with ports only.

File: docker-compose-installed.yaml (hunk 3 above)
Comment (cliveb) [reply docker-compose-installed.yaml]: Issue I'm solving is port 3030 connection refused issue on Seth. AFAIK Hyperledger/sawtooth-seth will not standup with current stable docker 18:03.1-ce without yaml files changes on Mac 10.13.5 or Ubuntu lts 16.04 to 18.04.

File: docker-compose-installed.yaml (hunk 1 above)
Comment (cliveb) [reply docker-compose-installed.yaml]: Please see my revised changes, tested and revised commit message.

Comment (dplumb94) [reply docker-compose-installed.yaml]: Compose file is invalid because you have the following structure:
```
args:
    args:
    ...
```

**Comment by knkski** [comment]
@cliveb: Did #38 fix the issue for this PR, or is this PR still necessary?

**Comment by knkski** [comment]
@cliveb: Thanks!
